haiku-apps/alivejournal/alivejournal-1.61.recipe

```diff
@@ -46,5 +46,6 @@ INSTALL()
 	addAppDeskbarSymlink $appsDir/AliveJournal
 	install -t $docDir README
 	sed -n '/http://www.bebits.com/app/1920/q;p' $docDir/README
+	rc -d -o ../../../additional-files/alivejournal.rdef work-$portVersion/sources/alivejournal-$gitSrcRevision/Resource.rsrc
 	addResourcesToBinaries $portDir/additional-files/alivejournal.rdef $appsDir/AliveJournal
```

Comment:
There are 3 scenarios:

1) if you only have the rsrc file and don't care about the rdef file:
```
	xres -o $appsDir/AliveJournal Resource.rsrc
```
(I'm not 100% sure if this works this way here.)

2) if you have the rdef and want to convert it to a rsrc:
```
	rc -o Resource.rsrc $portDir/additional-files/alivejournal.rdef
	xres -o $appsDir/AliveJournal Resource.rsrc
```

3) if you have the rdef in additional-files and want to let addResourcesToBinaries take care of everything:
```
	addResourcesToBinaries $portDir/additional-files/alivejournal.rdef \
		$appsDir/AliveJournal
```

Here I think you are trying to generate the rdef from the rsrc at build time.
We prefer to generate the rdef and add it to out repo in haiku-apps/alivejournal/additional-files/.

The best would be to ask upstream to host the rdef and make the changes to use it instead of the rsrc.

But as @scottmc wrote, you can (should?) first create the rdef and write the recipe using the third scenario described above. Then, when everything is OK, you can open a PR on the upstream repo with all the changes and a pointer to your recipe in the description, showing that it builds fine using the rdef you generated from the upstream rsrc.

Reply:
Ok.